nhrpd/nhrp_peer.c

```diff
@@ -76,7 +76,9 @@ static void __nhrp_peer_check(struct nhrp_peer *p)
 	unsigned online;
 
 	online = nifp->enabled && (!nifp->ipsec_profile || vc->ipsec);
-	if (p->online != online) {
+	/* TODO: nhrp_vc.c should be patched to ensure vc->ipsec == 0 when
+	 *       IPSec connection parameters are updated */
+	if ((p->online != online) || (p->t_fallback && !online) ) {
```

Comment:
that means patch is incomplete ? 
let see then.

Reply:
The patch is complete in that it fixes the problem that was seen (tunnel not getting re-established due to race condition). However this fix was a bit of a 'hack' to get a fix done for a release, and there is probably a more elegant solution. I will remove this part of the pull-request I think.

Reply:
This change was removed from latest version of code

Reply:
actually, I did not manage to understand this specific patch. as it has been removed, it is ok. lets continue discussing this item on a separate pull request.

Reply:
The issue here is that there is a race condition. The p->online variable is set to 1 in a timer event 50ms after this function. But if the __nhrp_peer_check() function is called and the online=0 within that 50ms, then the incorrect logic is taken.

I am not sure though that the way this is solved in this patch is actually the right way to handle it.